src/mpi/controller.rs

```diff
@@ -85,7 +110,18 @@ pub fn run(world: SystemCommunicator, config: Config) {
     let duration = end.sub(start).as_millis() / 1000;
     info!("#{rank} took: {duration}s");
 
+    info!("output dir: {:?}", config.output_dir);
+
+    if rank == 0 && config.routing_mode == RoutingMode::AdHoc {
```

Comment:
Do you only have a tmp dir for process 0? Or do all routers of all processes share the same tmp folder?

Reply:
THe folder structure is routing/<rank>/<temp_files>. And the root folder may only be deleted once